pkg/agent/proxy/proxier_test.go

```diff
@@ -216,6 +222,83 @@ func TestClusterIPv6(t *testing.T) {
 	testClusterIP(t, net.ParseIP("10:20::41"), net.ParseIP("10:180::1"), true)
 }
 
+func TestDualStackService(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	defer ctrl.Finish()
+	mockOFClient := ofmock.NewMockClient(ctrl)
+	fpv4 := NewFakeProxier(mockOFClient, false)
+	fpv6 := NewFakeProxier(mockOFClient, true)
+
```

Comment:
don't you think we should create a dual-stack proxier for this using these 2 fake proxiers and call `OnServiceUpdate`, etc. on the dual-stack proxier to make the test more accurate and exercise more code?

Reply:
Agree and updated.